sunscreen/src/types/zkp/native_field.rs

```diff
@@ -49,18 +49,78 @@ impl<F: BackendField> NativeField<F> {
     }
 }
 
-impl<F: BackendField, T> From<T> for NativeField<F>
-where
-    T: Into<BigInt>,
+impl<F: BackendField> From<u8> for NativeField<F>
 {
-    fn from(x: T) -> Self {
+    fn from(x: u8) -> Self {
+        (x as u64).into()
+        
+    }
+}
+
+impl<F: BackendField> From<u16> for NativeField<F>
+{
+    fn from(x: u16) -> Self {
+        (x as u64).into()
+        
+    }
+}
+
+impl<F: BackendField> From<u32> for NativeField<F>
+{
+    fn from(x: u32) -> Self {
+        (x as u64).into()
```

Comment:
Not a huge deal, but these `x as u64`s rely on the programmer verifying that `x` fits into `u64`. I think for this `From` instance `x.into()` is more appropriate, letting the compiler verify this instead.

Reply:
Fixed.